codesign/export.go

```diff
@@ -1,21 +1,106 @@
 package codesign
 
 import (
+	"bytes"
 	"errors"
 	"fmt"
+	"io/ioutil"
+	"os"
 	"path/filepath"
+	"strings"
 
+	"github.com/bitrise-io/codesigndoc/bitriseio"
+	"github.com/bitrise-io/codesigndoc/bitriseio/bitrise"
+	"github.com/bitrise-io/codesigndoc/models"
+	"github.com/bitrise-io/codesigndoc/osxkeychain"
+	"github.com/bitrise-io/codesigndoc/utility"
 	"github.com/bitrise-io/go-utils/command"
 	"github.com/bitrise-io/go-utils/log"
-	"github.com/bitrise-tools/codesigndoc/osxkeychain"
-	"github.com/bitrise-tools/go-xcode/certificateutil"
-	"github.com/bitrise-tools/go-xcode/profileutil"
+	"github.com/bitrise-io/go-xcode/certificateutil"
+	"github.com/bitrise-io/go-xcode/profileutil"
+	"github.com/bitrise-io/goinp/goinp"
 )
 
-// CollectAndExportIdentities exports the given certificates into the given directory as a single .p12 file
-func CollectAndExportIdentities(certificates []certificateutil.CertificateInfoModel, absExportOutputDirPath string, isAskForPassword bool) error {
+// UploadConfig contains configuration to automatically upload artifacts to bitrise.io
+type UploadConfig struct {
+	PersonalAccessToken string
+	AppSlug             string
+}
+
+func (config *UploadConfig) isValid() bool {
+	return (strings.TrimSpace(config.PersonalAccessToken) != "") &&
+		(strings.TrimSpace(config.AppSlug) != "")
+}
+
+// UploadAndWriteCodesignFiles exports then uploads codesign files to bitrise.io and saves them to output folder
+func UploadAndWriteCodesignFiles(certificates []certificateutil.CertificateInfoModel, profiles []profileutil.ProvisioningProfileInfoModel, askForPassword bool, outputDirPath string, uploadConfig UploadConfig) (bool, bool, error) {
+	identities, err := collectAndExportIdentities(certificates, askForPassword)
+	if err != nil {
+		return false, false, err
+	}
+
+	provisioningProfiles, err := collectAndExportProvisioningProfiles(profiles)
+	if err != nil {
+		return false, false, err
+	}
+
+	var client *bitrise.Client
+	if uploadConfig.isValid() {
```

Comment:
what if the user provided only the access token by an accident? i would log a warning that you need to use both flags together